library/src/main/java/com/datatorrent/lib/io/fs/AbstractThroughputFileInputOperator.java

```diff
@@ -122,8 +123,18 @@ public void setPartitionCount(int requiredPartitions)
   @Override
   public void emitTuples()
   {
-    scanDirectory();
-
+    int fileCt = 0;
```

Comment:
resetting fileCt to zero in emitTuple does not help, It needs to be cleared in beginWindow. as emitTuple can get called multipleTimes in a window.